browser/policy/brave_origin_policy_provider.cc

```diff
@@ -0,0 +1,216 @@
+      brave_wallet::prefs::kDisabledByPolicy, base::Value(true));
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveAIChatEnabled,
+      ai_chat::prefs::kEnabledByPolicy, base::Value(false));
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveSpeedreaderDisabled,
+      speedreader::kSpeedreaderDisabledByPolicy, base::Value(true));
+#endif
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveNewsDisabled,
+      brave_news::prefs::kBraveNewsDisabledByPolicy, base::Value(true));
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveVPNDisabled,
+      brave_vpn::prefs::kManagedBraveVPNDisabled, base::Value(true));
+#endif
+  SetBraveOriginPolicyWithPreference(
+      policy_map, policy_settings, policy::key::kBraveTalkDisabled,
+      kBraveTalkDisabledByPolicy, base::Value(true));
+
+  return bundle;
+}
+
+bool BraveOriginPolicyProvider::IsPolicySetByExternalProvider(
+    const std::string& policy_key) const {
+  // We need to check the global policy service to see if this policy
+  // is already set by other providers (registry, plist files, etc.)
+
+  if (!policy_service_) {
+    return false;
+  }
+
+  // Check if the policy is set by any provider
+  const policy::PolicyMap& policies = policy_service_->GetPolicies(
+      policy::PolicyNamespace(policy::POLICY_DOMAIN_CHROME, std::string()));
+
+  const policy::PolicyMap::Entry* entry = policies.Get(policy_key);
+  if (!entry) {
+    return false;
+  }
+
+  // Check if the source is something other than our BraveOrigin source
+  // If it's set by registry, plist, etc., we should respect that
+  return entry->source != policy::POLICY_SOURCE_ENTERPRISE_DEFAULT;
+}
+
+void BraveOriginPolicyProvider::SetBraveOriginPolicyWithPreference(
+    policy::PolicyMap& policy_map,
+    const base::Value::Dict& policy_settings,
+    const std::string& policy_key,
+    const std::string& pref_name,
+    const base::Value& default_value) {
+  // Only set policy if not already set by external providers
+  if (!IsPolicySetByExternalProvider(policy_key)) {
+    // Check if user has a stored preference for this policy
+    const base::Value* user_value = policy_settings.Find(pref_name);
+    const base::Value& value_to_use = user_value ? *user_value : default_value;
+
+    policy_map.Set(policy_key, policy::POLICY_LEVEL_MANDATORY,
+                   policy::POLICY_SCOPE_MACHINE,
+                   policy::POLICY_SOURCE_ENTERPRISE_DEFAULT,
+                   value_to_use.Clone(), nullptr);
+
+    // Track the preference that this policy controls in BraveOriginState
+    BraveOriginState* brave_origin_state = BraveOriginState::GetInstance();
+    if (brave_origin_state) {
+      brave_origin_state->AddBraveOriginControlledPref(pref_name);
```

Comment:
Alternatively, we could pass the policy key instead of the pref name, and allow `BraveOriginState` to lookup the pref name. This would remove some of the deps in this target.